.github/workflows/publish.yml

```
@@ -0,0 +1,117 @@
+# - These secrets must be set up in the repository settings:
+#   - SSH_PRIVATE_KEY: Private SSH key for accessing the vps
+#   - SSH_USER: The SSH username for the vps
+#   - SSH_HOST: The SSH host (IP or domain) of the vps
+#   - WORK_DIR: The directory on the vps where the app is/will be deployed
+#   - BOT_TOKEN_{Environment}_{BranchName}: Environment variable for the bot
+#     - Example: BOT_TOKEN_PRODUCTION_master
+#     - Environment default: PRODUCTION
+#   - MONGO_URI: Environment variable for the bot
+#   - HYPIXEL_KEY: Environment variable for the bot
+#   - CONFIG_FOLDER_PATH: Path to the config file folder on the vps
+
+name: Publish
+
+on:
+  workflow_dispatch:
+    inputs:
+      botEnvironmentType:
+        description: 'Bot environment type (All uppercase):'
+        required: true
+        default: 'PRODUCTION'
+        type: string
+      port:
+        description: 'Port:'
+        required: true
+        default: '8080'
+        type: string
+      useOldImage:
+        description: 'Use old Docker image:'
+        required: true
+        default: false
+        type: boolean
+
+  push:
+    branches:
+      - 'master'
+      - 'generator'
+    paths-ignore:
+      - '**.md'
+      - '.github/**'
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.actor }}/nerd-bot-${{ github.event.inputs.botEnvironmentType || 'PRODUCTION' }}-${{ github.ref_name }}:latest
+  CONTAINER_NAME: nerd-bot-${{ github.event.inputs.botEnvironmentType || 'PRODUCTION' }}-${{ github.ref_name }}
+  ENVIRONMENT_NAME: ${{ github.event.inputs.botEnvironmentType || 'PRODUCTION' }}
+  PORT: ${{ github.event.inputs.port || '1234' }} # Default port is 1234 but should probably be changed to 8080 (convention)
+  NAME_BOT_TOKEN_SECRET: BOT_TOKEN_${{ github.event.inputs.botEnvironmentType || 'PRODUCTION' }}_${{ github.ref_name }}
+  USE_OLD_IMAGE: ${{ github.event.inputs.useOldImage || 'false' }}
+
+jobs:
+  publish:
+    name: Publish image
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        if: ${{ env.USE_OLD_IMAGE == 'false' }}
+        uses: actions/checkout@v4.2.2
+
+      - name: Login to GHCR
+        if: ${{ env.USE_OLD_IMAGE == 'false' }}
+        run: echo "${{ secrets.GITHUB_TOKEN }}" | docker login ${{ env.REGISTRY }} -u ${{ github.actor }} --password-stdin
+
+      - name: Build and push Docker image
+        if: ${{ env.USE_OLD_IMAGE == 'false' }}
+        run: |
+          docker build . --tag ${{ env.REGISTRY }}/$(echo "${{ env.IMAGE_NAME }}" | tr '[:upper:]' '[:lower:]') \
+            --build-arg GITHUB_USERNAME=${{ secrets.GITHUB_USERNAME }} \
+            --build-arg GITHUB_TOKEN=${{ secrets.GITHUB_TOKEN }} \
+            --build-arg REPO_USERNAME=${{ env.REPO_USERNAME }} \
+            --build-arg REPO_NAME=${{ env.REPO_NAME }} \
+            --build-arg REPO_BRANCH=${{ env.REPO_BRANCH }}
+
+          docker push ${{ env.REGISTRY }}/$(echo "${{ env.IMAGE_NAME }}" | tr '[:upper:]' '[:lower:]')
```

```suggestion
          docker build . --tag ${{ env.REGISTRY }}/$(echo "${{ env.IMAGE_NAME }}" | tr '[:upper:]' '[:lower:]') 
          docker push ${{ env.REGISTRY }}/$(echo "${{ env.IMAGE_NAME }}" | tr '[:upper:]' '[:lower:]')
```

Old one had bug i just realized so here is proper one 

Still the adjustments needed to be able to merged #382